psh/test-kill.py

```diff
@@ -0,0 +1,145 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# psh kill command test
+#
+# Copyright 2021 Phoenix Systems
+# Author: Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+from psh.tools.basic import run_psh, assert_only_prompt
+from time import sleep
+
+PROMPT = r'\r\x1b\[0J' + r'\(psh\)% '
+EOL = r'(\r+)\n'
+
+
+def is_hardware_target(p):
+    cmd = 'ls'
+
+    p.sendline(cmd)
+    p.expect(cmd)
+    idx = p.expect([
+            'syspage',
+            'bin'])
+    p.expect(PROMPT)
+
+    if idx == 0:
+        return True
+    elif idx == 1:
+        return False
+
+
+def get_process_list(p):
+    header_seen = False
+    ps_header = 'PID PPID PR STATE %CPU WAIT TIME VMEM THR CMD'.split()
+    ps_list = []
+    oneline_pattern = r'(.*?)(\r+)\n'
+    cmd = 'ps'
+    p.sendline(cmd)
+    p.expect(cmd + EOL)
+    while True:
+        idx = p.expect([PROMPT, oneline_pattern])
+        if idx == 0:
+            break
+        line = p.match.group(1)
+        if not header_seen and line.split() == ps_header:
+            header_seen = True
+        else:
+            try:
+                pid, ppid, pr, state, cpu, wait, time, vmem, thr, task = line.split()
+            except ValueError:
+                assert False, f'wrong ps output: {line}'
+
+            ps_list.append({'pid': pid, 'task': task, 'state': state})
```

Comment:
Proposition 1:
```suggestion
        if not header_seen:
            line = line.strip()
            header_seen = line.startswidth('PID')
        else:
            try:
                pid, ppid, pr, state, cpu, wait, time, vmem, thr, task = line.split(maxsplit=9)
            except ValueError:
                assert False, f'wrong ps output: {line}'

            ps_list.append({'pid': pid, 'task': task, 'state': state})
```
The variable `ps_header` is not needed.

Proposition 2:
```suggestion
        try:
            pid, ppid, pr, state, cpu, wait, time, vmem, thr, task = line.split(maxsplit=9)
        except ValueError:
            assert False, f'wrong ps output: {line}'

        if pid.isdigit():
            ps_list.append({'pid': pid, 'task': task, 'state': state})
```
The variables `ph_header` and `header_seen` are not necessary.
`maxsplit=9` is needed when `cmd` is used with arguments.
Maybe `get_process_list(p)` define as a `generator`.

Reply:
You are right, thanks